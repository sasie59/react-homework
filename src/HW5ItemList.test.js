import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import HW from './HW5ItemList';

test('should HW5ItemList', () => {
  const { container } = render(<HW />);
  expect(container).toMatchSnapshot();
});

test('should add something to list', () => {
  const { container } = render(<HW />);
  const input = container.querySelector('input');

  userEvent.type(input, "hooks{enter}");

  expect(container).toMatchSnapshot();
});
test('should add something to list', () => {
  const { container } = render(<HW />);
  const input = container.querySelector('input');

  userEvent.type(input, "react{enter}");
  userEvent.type(input, "jest{enter}");

  expect(container).toMatchSnapshot();
});

