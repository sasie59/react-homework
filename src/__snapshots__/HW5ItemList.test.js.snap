// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should HW5ItemList 1`] = `
<div>
  <div>
    <h1>
      HW5ItemList
    </h1>
    <form
      action=""
    >
      <input
        type="text"
      />
    </form>
    <ul />
  </div>
</div>
`;

exports[`should add something to list 1`] = `
<div>
  <div>
    <h1>
      HW5ItemList
    </h1>
    <form
      action=""
    >
      <input
        type="text"
      />
    </form>
    <ul>
      <li>
        hooks
      </li>
    </ul>
  </div>
</div>
`;

exports[`should add something to list 2`] = `
<div>
  <div>
    <h1>
      HW5ItemList
    </h1>
    <form
      action=""
    >
      <input
        type="text"
      />
    </form>
    <ul>
      <li>
        jest
      </li>
      <li>
        react
      </li>
    </ul>
  </div>
</div>
`;
